src/axom/sidre/core/Group.hpp

```diff
@@ -1242,10 +1262,29 @@ class Group
    * same name as given Group, method is a no-op.
    *
    * \return pointer to given argument Group object or nullptr if Group
-   * is not moved into this Group.
+   * is not copied into this Group.
    */
   Group* copyGroup(Group* group);
 
+  /*!
+   * \brief Create a deep copy of Group hierarchy rooted at given Group and
+   *        make it a child of this Group.
+   *
+   * Note that all Views in the Group hierarchy are deep copied as well.
```

Comment:
Should we add a default bool arg to this; e.g., copy_child_groups = true?  This would have the same interface, but would allow users more control if they needed it.

Reply:
I tried to be consistent with the API used in the existing `copyGroup()` method.  A Group is a hierarchical container containing Groups and Views, and I think the expected behavior of copying a container is that you copy all of its contents.